### PR TITLE
Extend update instructions to be able to preserve updated apps when installing bundles

### PR DESCRIPTION
**Pull request by PVince81**
Fixes https://github.com/owncloud/docs/issues/828

See the ticket for context.

This is to prevent enterprise-complete bundles to overwrite already upgraded versions with older versions, as it could happen with the upcoming 10.1.1 hotfix release.

### CODE REVIEW

File: modules/admin_manual/pages/maintenance/manual_upgrade.adoc [hunk 1]
```diff
@@ -146,6 +134,32 @@ code) and the Market app is enabled, but there is no available internet
 connection, then these apps will need to be manually updated once the
 upgrade is finished.
 
+[[copy-old-updated-apps]]
+== Copy Old Updated Apps
+
+If your setup has a read-only `apps/` folder and is using a separate
```
Comment (mmattel) [inline modules/admin_manual/pages/maintenance/manual_upgrade.adoc]: Why not writing:
```
You can skip this section if:
- Your setup has ...
- You are installing
```

File: modules/admin_manual/pages/maintenance/manual_upgrade.adoc [hunk 2]
```diff
@@ -146,6 +134,32 @@ code) and the Market app is enabled, but there is no available internet
...
+This could happen for example when installing a patch or hotfix bundle in which
+only minimal changes are contained without recent app updates that
+exist on the marketplace.
+
+For every app that has an older version in the bundle
+compared to the formerly installed version, please delete the one that
+came with the bundle and replace it with the one from your old `apps/`
+folder. This will make sure that there are no version discrepancies.
+
+Note that this procedure does not apply for third party apps which you are
```
Comment (mmattel) [inline modules/admin_manual/pages/maintenance/manual_upgrade.adoc]: Better:
```
NOTE: This procedure does 
```
Which renders to a note icon + text

**Comment by PVince81** [comment]
@mmattel updated

**Comment by mmattel** [comment]
There is no commit nor change I can see?

File: modules/admin_manual/pages/maintenance/manual_upgrade.adoc [hunk 3]
```diff
@@ -18,6 +18,12 @@ First, xref:maintenance/backup.adoc[backup] the following items:
 cp -rv /var/www/owncloud /opt/backup/owncloud && mysqldump <db_name> > /opt/backup/backup-file.sql
 ....
 
+[[review-installed-apps]]
```
Comment (settermjd) [inline modules/admin_manual/pages/maintenance/manual_upgrade.adoc]: There's no need to explicitly add a header as one, exactly the same, will be auto-generated.

File: modules/admin_manual/pages/maintenance/manual_upgrade.adoc [hunk 4]
```diff
@@ -18,6 +18,12 @@ First, xref:maintenance/backup.adoc[backup] the following items:
...
+== Review Installed Apps
+
+Take note of apps that are currently installed and their versions by
+running `occ apps:list` and saving it for future reference.
```
Comment (settermjd) [inline modules/admin_manual/pages/maintenance/manual_upgrade.adoc]: s/saving/save/g

File: modules/admin_manual/pages/maintenance/manual_upgrade.adoc [hunk 5]
```diff
@@ -146,6 +134,32 @@ code) and the Market app is enabled, but there is no available internet
 connection, then these apps will need to be manually updated once the
 upgrade is finished.
 
+[[copy-old-updated-apps]]
```
Comment (settermjd) [inline modules/admin_manual/pages/maintenance/manual_upgrade.adoc]: No need to explicitly create an anchor.

File: modules/admin_manual/pages/maintenance/manual_upgrade.adoc [hunk 6]
```diff
@@ -146,6 +134,32 @@ code) and the Market app is enabled, but there is no available internet
...
+== Copy Old Updated Apps
+
+If your setup has a read-only `apps/` folder and is using a separate
+writable folder for updated apps like `apps-external/` or another location,
+please skip this section as well.
+
+If you are installing the standard owncloud community bundle, usually
```
Comment (settermjd) [inline modules/admin_manual/pages/maintenance/manual_upgrade.adoc]: s/owncloud/ownCloud/g

Comment (settermjd) [reply modules/admin_manual/pages/maintenance/manual_upgrade.adoc]: Plus, it's likely good to draw attention to this and the paragraph above by using admonitions.

File: modules/admin_manual/pages/maintenance/manual_upgrade.adoc [hunk 7]
```diff
@@ -162,7 +176,8 @@ The upgrade operation can take anywhere from a few minutes to a few hours, depen
 When it is finished you will see either a success message, or an error message which indicates why the process did not complete successfully.
 
 [[copy-old-apps]]
-== Copy Old Apps
+[[copy-old-third-party-apps]]
```
Comment (settermjd) [inline modules/admin_manual/pages/maintenance/manual_upgrade.adoc]: An explicit anchor isn't required.

File: modules/admin_manual/pages/maintenance/manual_upgrade.adoc (hunk 3 above)
Comment (PVince81) [reply modules/admin_manual/pages/maintenance/manual_upgrade.adoc]: I know... for some reason this file already had those so I thought it was an expected pattern

File: modules/admin_manual/pages/maintenance/manual_upgrade.adoc (hunk 6 above)
Comment (PVince81) [reply modules/admin_manual/pages/maintenance/manual_upgrade.adoc]: @settermjd not sure how to do the admonition part. Add an addition NOTE: block at the beginning of the section ? What to write to draw attention ?

File: modules/admin_manual/pages/maintenance/manual_upgrade.adoc (hunk 3 above)
Comment (PVince81) [reply modules/admin_manual/pages/maintenance/manual_upgrade.adoc]: also I think it's nice to keep these even when the title changes to make sure that older links still work. or is there another way ?

**Comment by PVince81** [comment]
I've fixed only the "owncloud" one. I'd leave the anchors items as they were there before.
If we want to discard this pattern then we should go through the whole doc and remove all redundant anchors.

Regarding the extra admonition, please add it yourself @settermjd. Feel free to reword the other sections if needed.

File: modules/admin_manual/pages/maintenance/manual_upgrade.adoc (hunk 3 above)
Comment (settermjd) [reply modules/admin_manual/pages/maintenance/manual_upgrade.adoc]: Some do have them, though it's my beliefe that they should be removed. I believe a significant number of them were auto-generated during the conversion from sphinx-doc with Pandoc. There's nothing wrong with keeping them. It just seemed superfluous (in this instance) if one that's exactly the same would be auto-generated.

**Comment by PVince81** [comment]
@mmattel seems I missed pushing.

@settermjd please check the latter PR

File: modules/admin_manual/pages/maintenance/manual_upgrade.adoc (hunk 6 above)
Comment (settermjd) [reply modules/admin_manual/pages/maintenance/manual_upgrade.adoc]: [Here's the docs for admonitions](https://asciidoctor.org/docs/user-manual/#admonition).

Comment (mmattel) [reply modules/admin_manual/pages/maintenance/manual_upgrade.adoc]: or here in a condensed version :smile:
https://github.com/owncloud/docs/blob/master/docs/best-practices.md#admonition

**Comment by PVince81** [comment]
this PR originated to the fact that we intended to bundle old apps with 10.1.1 enterprise complete tarball to avoid breaking semver.

we'll be reworking the bundling concept and what they mean, as the above approach is too complicated for admins.

closing as obsolete

**Pull request closed without merging**